lib/Recur/RRuleIterator.php

```diff
@@ -1230,4 +1238,103 @@ protected function getMonths(): array
 
         return $recurrenceMonths;
     }
+
+    private function getNextIterationDateTime(\DateTimeInterface $dateTime, string $modify): \DateTimeInterface
+    {
+        $startTs = $dateTime->getTimestamp();
+        $initialDateTime = clone $dateTime;
+        $modifiedDateTime = $dateTime->modify($modify);
+
+        $this->detectAndSaveDstOffset(
+            $initialDateTime,
+            $modifiedDateTime,
+            $modify,
+        );
+
+        $modifiedDateTime = $this->revertPastOffset($modifiedDateTime);
```

Comment:
why not call this at the start of the loop, it seems error-prone to me to call it in getNextIterationDateTime (never mind the fact that the name implies a readonly function)

Reply:
The issue is that if we revert at the start we fall back on the dst leap and the revert is effectively canceled